week2/starter/ColorPicker/ContentView.swift

```diff
@@ -33,54 +33,60 @@
 import SwiftUI
 
 struct ContentView: View {
+  @Environment(\.verticalSizeClass) var verticalSizeClass
+  @Environment(\.horizontalSizeClass) var horizontalSizeClass
+
+  
   @State private var alertIsVisible: Bool = false
-  @State private var redColor: Double = 0.0
-  @State private var greenColor: Double = 0.0
-  @State private var blueColor: Double = 0.0
-  @State private var foregroundColor = Color(red: 0, green: 0, blue: 0)
+  @State private var redColor: Double = 255.0
+  @State private var greenColor: Double = 100.0
+  @State private var blueColor: Double = 50.0
+  @State private var foregroundColor = Color(red: 255.0/Constants.General.maxValueForColor, green: 100.0/Constants.General.maxValueForColor , blue: 50.0/Constants.General.maxValueForColor)
 
   var body: some View {
-
-    VStack {
-      Text("Color Picker")
-        .font(.largeTitle)
-
-      RoundedRectangle(cornerRadius: 0)
-        .foregroundColor(foregroundColor)
-        .border(.black)
-      VStack {
-        Text("Red")
-        HStack {
-          Slider(value: $redColor, in: 0...255)
-          Text("\(Int(redColor.rounded()))")
-        }
-      }
-      VStack {
-        Text("Green")
-        HStack {
-          Slider(value: $greenColor, in: 0...255)
-          Text("\(Int(greenColor.rounded()))")
-        }
-      }
+    
       VStack {
-        Text("Blue")
-        HStack {
-          Slider(value: $blueColor, in: 0...255)
-          Text("\(Int(blueColor.rounded()))")
+        TextView()
+        if verticalSizeClass == .regular && horizontalSizeClass == .compact  {
+          
+            RectangleView(foregroundColor: $foregroundColor)
+            SliderViews(red: $redColor, green: $greenColor, blue: $blueColor)
+        }else {
+          HStack{
+            RectangleView(foregroundColor: $foregroundColor)
+            VStack{
```

Comment:
It's better to move this VStack inside `SliderViews`. At first, you might think "but we can't do that because in portrait mode it will be a VStack inside another VStack". But that wouldn't be a problem.